apps/spikeConverter.cpp

```diff
@@ -29,11 +43,52 @@ int main( int argc, char* argv[] )
     }
 
     lunchbox::Clock clock;
-    const brion::SpikeReport in( brion::URI( argv[1] ), brion::MODE_READ );
+    brion::SpikeReport in( brion::URI( argv[1] ), brion::MODE_READ );
     const float readTime = clock.resetTimef();
 
     brion::SpikeReport out( brion::URI( argv[2] ), brion::MODE_WRITE );
-    out.writeSpikes( in.getSpikes( ));
+
+    if( in.getReadMode() == brion::SpikeReport::STREAM )
+    {
+        // Stream-to-File conversion
+        while( in.getNextSpikeTime() != brion::UNDEFINED_TIMESTAMP )
+        {
+            in.waitUntil( brion::UNDEFINED_TIMESTAMP, STREAM_READ_TIMEOUT_MS );
+            out.writeSpikes( in.getSpikes( ));
+        }
```

Comment:
in.waitUntil( brion::UNDEFINED_TIMESTAMP ) is the way to say wait until the end of the stream.
in.getNextSpikeTime returns brion::UNDEFINED_TIMESTAMP when the end has been reached.
So what you should be reading is

   while not finished
       wait until end with timeout
       write current.

The mistake I see is that writeSpikes is going to repeat all previously read spikes every iteration (as in 1 2 1 2 3 4 1 2 3 4 5 6...).  To way to avoid it is to call clear after each writeSpikes call.

A more clear way of doing the same would be providing new functions, but it isn't worth doing it if we plan to change the whole API to replace the std::multimap with a custom containers with magic iterators or something really low level. I didn't add those functions from the beginning because they are sort of redundant and preferred to keep the API lean.
I'm reluctant to write a low level API and prefer the iterator based one because I think that we are not going to get the low level API right for the stream and file based cases combined.